app/scripts/views/mixins/password-strength-mixin.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'lib/promise',
+  'lib/runtime-require',
+  'lib/url'
+], function (p, runtimeRequire, Url) {
+  'use strict';
+
+  var PasswordStrengthMixin = {
+    initialize: function (options) {
+      this._able = options.able;
+    },
+
+    _isPasswordStrengthCheckEnabledValue: undefined,
+    _isPasswordStrengthCheckEnabled: function () {
+      if (typeof this._isPasswordStrengthCheckEnabledValue === 'undefined') {
+        var abData = {
+          isMetricsEnabledValue: this.metrics.isCollectionEnabled(),
+          uniqueUserId: this.user.get('uniqueUserId'),
+          // the window parameter will override any ab testing features
+          forcePasswordStrengthCheck: Url.searchParam('passwordStrengthCheck', this.window.location.search)
+        };
+
+        this._isPasswordStrengthCheckEnabledValue =
+              this._able.choose('passwordStrengthCheckEnabled', abData);
+
+        if (this._isPasswordStrengthCheckEnabledValue) {
+          this._logStrengthExperimentEvent('enabled');
+        }
+      }
+      return this._isPasswordStrengthCheckEnabledValue;
+    },
+
+    _passwordStrengthCheckerPromise: undefined,
+    _getPasswordStrengthChecker: function () {
+      // returns a promise that resolves once the library is loaded.
+      var self = this;
+      if (! self._passwordStrengthCheckerPromise) {
+        self._passwordStrengthCheckerPromise = runtimeRequire('passwordcheck')
```

Comment:
requirejs has a default timeout for loading the resource ("The default is 7 seconds."), `runtimeRequire` may timeout and fail here, should we catch, log or do something about that?